backend-test/TestHelpers.hs

```diff
@@ -34,125 +34,49 @@ import System.Directory (removeFile)
 import System.Environment (setEnv, unsetEnv)
 import Test.Tasty (TestTree, testGroup, withResource)
 
--- | A minimal mock request for running a ServerPart
-mockRequest :: IO Request
-mockRequest = do
-    inputsBody <- newMVar []
-    requestBody <- newEmptyMVar
-    return Request
-        { rqSecure          = False
-        , rqMethod          = GET
-        , rqPaths           = []
-        , rqUri             = "/"
-        , rqQuery           = ""
-        , rqInputsQuery     = []
-        , rqInputsBody      = inputsBody
-        , rqCookies         = []
-        , rqVersion         = HttpVersion 1 1
-        , rqHeaders         = Map.empty
-        , rqBody            = requestBody
-        , rqPeer            = ("127.0.0.1", 0)
-        }
+-- | Generalized function to create a mock request
+createMockRequest :: Method -> String -> [(String, String)] -> BSL8.ByteString -> IO Request
+createMockRequest reMethod reUri queryInputs body = do
```

Comment:
Oh I'm not sure if I made a typo earlier, but you should use `rqMethod`/`rqUri` etc. for the argument names, not `reMethod`/`reqUri`.

I'm not sure if you're seeing a "name shadowing" warning; if so, use `reqMethod`/`reqUri`/etc. instead.